rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -536,14 +548,21 @@ typedef struct CustomParticipantInfo
   rmw_guard_condition_t * graph_guard_condition;
 } CustomParticipantInfo;
 
-extern "C"
+inline
+std::string
+_filter_ros_prefix(const std::string & topic_name)
 {
-// static for internal linkage
-static const char * const eprosima_fastrtps_identifier = "rmw_fastrtps_cpp";
-static const char * const ros_topic_prefix = "rt";
-static const char * const ros_service_requester_prefix = "rq";
-static const char * const ros_service_response_prefix = "rr";
+  auto prefixes = {ros_topic_prefix, ros_service_requester_prefix, ros_service_response_prefix};
+  for (auto prefix : prefixes) {
+    if (topic_name.rfind(std::string(prefix) + "/") == 0) {
```

Comment:
As we discussed on chat, this should be:

`if (topic_name.rfind(std::string(prefix) + "/", 0) == 0) {`